data/beaconrestapi/src/main/java/tech/pegasys/artemis/beaconrestapi/BeaconRestApi.java

```diff
@@ -126,12 +113,17 @@ private void addNodeHandlers(DataProvider provider) {
         new NodeSyncingHandler(provider.getSyncDataProvider(), jsonProvider));
   }
 
-  private void addBeaconHandlers(
-      ChainStorageClient chainStorageClient, CombinedChainDataClient combinedChainDataClient) {
-    ChainDataProvider provider = new ChainDataProvider(chainStorageClient, combinedChainDataClient);
+  private void addBeaconHandlers(final DataProvider dataProvider) {
+    final ChainStorageClient chainStorageClient = dataProvider.getChainStorageClient();
+    final CombinedChainDataClient combinedChainDataClient =
+        dataProvider.getCombinedChainDataClient();
```

Comment:
ChainDataProvider is part of dataProvider (dataProvider.getChainDataProvider), so you can just use that...